aiida_mlip/workflows/ht_workgraph.py

```diff
@@ -0,0 +1,126 @@
+"""Workgraph to run high-throughput calculations."""
+
+from pathlib import Path
+from typing import Callable, Union
+
+from aiida.engine import CalcJob, WorkChain
+from aiida_workgraph import WorkGraph, task
+from ase.io import read
+
+from aiida_mlip.helpers.help_load import load_structure
+
+
+@task.graph_builder(outputs=[{"name": "final_structures", "from": "context.structs"}])
+def ht_calc_builder(
+    calc: Union[CalcJob, Callable, WorkChain, WorkGraph],
+    folder: Path,
+    calc_inputs: dict,
+    input_struct_key: str = "struct",
+    final_struct_key: str = "final_structure",
+) -> WorkGraph:
+    """
+    Build high throughput calculation WorkGraph.
+
+    The `calc` must take a structure, by default `struct`, as one of its inputs.
+    Tasks will then be created to carry out the calculation for each structure file in
+    `folder`.
+
+    Parameters
+    ----------
+    calc : Union[CalcJob, Callable, WorkChain, WorkGraph]
+        Calculation to be performed on all structures.
+    folder : Path
+        Path to the folder containing input structure files.
+    calc_inputs : dict
+        Dictionary of inputs, shared by all the calculations. Must not contain
+        `struct_key`.
+    input_struct_key : str
+        Keyword for input structure for `calc`. Default is "struct".
+    final_struct_key : str
+        Key for final structure output from `calc`. Default is "final_structure".
+
+    Returns
+    -------
+    WorkGraph
+        The workgraph with calculation tasks for each structure.
+
+    Raises
+    ------
+    FileNotFoundError
+        If `folder` has no valid structure files.
+    """
+    wg = WorkGraph()
+    structure = None
+
+    for child in folder.glob("**/*"):
```

Comment:
```suggestion
    for child in folder.glob("**/*"):
```
`folder.walk()`?

Reply:
This is only available from Python 3.12 onwards

Reply:
Weird, so it is. I hate Python versioning. In any case you probably want
```python
    for child in folder.rglob("*"):
```
possibly with
```python
    for child in (for file in folder.rglob("*") if file.is_file()):
```

Reply:
Does `rglob` essentially just add the `**/`? I was thinking we may want an option to disable recursive search, so might it make more sense to use glob with/without that?

Agree that checking the file makes sense, thanks

Reply:
Re `rglob`: Yeah, basically.